ThirdPartyAdapters/unity/unity/src/main/java/com/google/ads/mediation/unity/UnityRewardedAd.java

```diff
@@ -199,45 +158,83 @@ public void showAd(Context context) {
     public void onUnityAdsShowStart(String placementId) {
       // Unity Ads video ad started playing. Send Video Started event if this is a rewarded
       // video.
-      if (mMediationRewardedAdCallback != null) {
-        mMediationRewardedAdCallback.reportAdImpression();
-        mMediationRewardedAdCallback.onVideoStart();
-      }
+      sendRewardedAdEvent(AdEvent.IMPRESSION);
+      sendRewardedAdEvent(AdEvent.VIDEO_START);
     }
 
     @Override
     public void onUnityAdsShowClick(String placementId) {
       // Unity Ads ad clicked.
-      if (mMediationRewardedAdCallback != null) {
-        mMediationRewardedAdCallback.reportAdClicked();
-      }
+      sendRewardedAdEvent(AdEvent.CLICK);
     }
 
     @Override
     public void onUnityAdsShowComplete(String placementId,
         UnityAds.UnityAdsShowCompletionState state) {
       // Unity Ads ad closed.
-      if (mMediationRewardedAdCallback == null) {
-        return;
-      }
-
       // Reward is provided only if the ad is watched completely.
       if (state == UnityAds.UnityAdsShowCompletionState.COMPLETED) {
-        mMediationRewardedAdCallback.onVideoComplete();
-        // Unity Ads doesn't provide a reward value. The publisher is expected to
-        // override the reward in AdMob console.
-        mMediationRewardedAdCallback.onUserEarnedReward(new UnityReward());
+        sendRewardedAdEvent(AdEvent.COMPLETE);
+        sendRewardedAdEvent(AdEvent.REWARD);
       }
-      mMediationRewardedAdCallback.onAdClosed();
+      sendRewardedAdEvent(AdEvent.CLOSE);
     }
 
     @Override
     public void onUnityAdsShowFailure(String placementId, UnityAdsShowError error, String message) {
       // Unity Ads ad failed to show.
-      AdError adError = createSDKError(error, message);
       if (mMediationRewardedAdCallback != null) {
+        AdError adError = createSDKError(error, message);
         mMediationRewardedAdCallback.onAdFailedToShow(adError);
       }
     }
   };
+
+  private void sendRewardedLoadSuccess() {
+    if (mMediationAdLoadCallback != null) {
+      mMediationRewardedAdCallback = mMediationAdLoadCallback.onSuccess(UnityRewardedAd.this);
+    }
+  }
+
+  private void sendRewardedLoadFailure(AdError adError) {
+    Log.e(TAG, "Failed to load rewarded ad: " + adError.toString());
+    if (mMediationAdLoadCallback != null) {
+      mMediationAdLoadCallback.onFailure(adError);
+    }
+  }
+
+  private void sendRewardedAdEvent(AdEvent adEvent) {
+    if (mMediationRewardedAdCallback == null) {
+      return;
+    }
+
+    switch (adEvent) {
+      case OPEN:
+        mMediationRewardedAdCallback.onAdOpened();
+        break;
+      case IMPRESSION:
+        mMediationRewardedAdCallback.reportAdImpression();
+        break;
+      case VIDEO_START:
+        mMediationRewardedAdCallback.onVideoStart();
+        break;
```

Comment:
I see what you did here, good approach. You could go a bit further(the same way we did on iOS).
There is slight difference between Interstitial and rewarded handlers, you could create a class that accepts either reward or interstitial listener and combine common logic for handling events over there. In this case you would have cleaner codebase in Ads objects. @vita-unity can help you with that.

Reply:
Done and tested @alex-crowe, let me know what you think :)